### PR TITLE
LPS-133410 | master

### PR DESCRIPTION
**Pull request by ozysouza**
https://issues.liferay.com/browse/LPS-133410

### CODE REVIEW

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/categories/CategoriesUseCase.testcase [hunk 1]
```diff
@@ -1272,6 +1272,56 @@ definition {
...
+
+		task ("Click on the plus button to add a vocabulary") {
+			Category.openCategoriesAdmin(siteURLKey = "test-site-name");
+
+			Click(locator1 = "NavNested#NAV_NESTED_PLUS");
+		}
+
+		task ("View the alert not shown") {
+			AssertAlertNotPresent();
+		}
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/categories/CategoriesUseCase.testcase:1322]:
```suggestion
Vocabulary. gotoAddVocabulary
```

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/categories/CategoriesUseCase.testcase [hunk 2]
```diff
@@ -1272,6 +1272,56 @@ definition {
...
+				fieldName = "Text");
+
+			WebContentStructures.saveCP();
+		}
+
+		task ("View the XSS is not escaped") {
+			LexiconEntry.viewEntryName(rowEntry = "${structureName}");
+		}
+
+		task ("Click on the plus button to add a vocabulary") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/categories/CategoriesUseCase.testcase:1314]: View no alert shown when create a vocabulary

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/categories/CategoriesUseCase.testcase [hunk 3]
```diff
@@ -1272,6 +1272,56 @@ definition {
...
+			Category.openCategoriesAdmin(siteURLKey = "test-site-name");
+
+			Click(locator1 = "NavNested#NAV_NESTED_PLUS");
+		}
+
+		task ("View the alert not shown") {
+			AssertAlertNotPresent();
+		}
+	}
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/categories/CategoriesUseCase.testcase:1323]: Step into Vocabulary.gotoAddVocabulary already cover check alert. selenium.assertJavaScriptErrors();

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/categories/CategoriesUseCase.testcase [hunk 4]
```diff
@@ -1272,6 +1272,56 @@ definition {
...
 	}
 
+	@description = "This is a test for LPS-121450.  Can add a vocabulary without recieve an alert after set DM and WC Strucutre with XSS name."
+	@priority = "4"
+	test ViewNoAlertWhenClickOnPlusButtonToAddVocabulary {
+		var documentName = '''<script>alert(123)</script>''';
+
+		var structureName = '''<script>alert(456)</script>''';
+
+		task ("Add a document type") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/categories/CategoriesUseCase.testcase:1282]: Add a document type with XSS name

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/categories/CategoriesUseCase.testcase [hunk 5]
```diff
@@ -1272,6 +1272,56 @@ definition {
...
+			DMDocumentType.add(
+				dmDocumentTypeFieldNames = "Text",
+				dmDocumentTypeName = "${documentName}");
+		}
+
+		task ("View the XSS is not escaped") {
+			LexiconEntry.viewEntryName(rowEntry = "${documentName}");
+		}
+
+		task ("Add a web content strucuture") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/categories/CategoriesUseCase.testcase:1296]: Add a web content structure with XSS name

**Review by Tim-Cao — COMMENTED** [review]
Hi @ozysouza, Could you update and move the case to CategoriesWithXSS.testcase? Thanks

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/categories/CategoriesUseCase.testcase [hunk 6]
```diff
@@ -1272,6 +1272,56 @@ definition {
...
+		}
+
+		task ("View the XSS is not escaped") {
+			LexiconEntry.viewEntryName(rowEntry = "${structureName}");
+		}
+
+		task ("Click on the plus button to add a vocabulary") {
+			Category.openCategoriesAdmin(siteURLKey = "test-site-name");
+
+			Click(locator1 = "NavNested#NAV_NESTED_PLUS");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/categories/CategoriesUseCase.testcase:1317]: Vocabulary. gotoAddVocabulary

**Comment by ozysouza** [comment]
Hi @Tim-Cao, PR Updated following your comments.

**Comment by Tim-Cao** [comment]
Hi @ozysouza, see changes https://github.com/liferay-echo/liferay-portal/compare/888fea3b2cf49efda2e69373a8a6e1b7c76c562f...8b35192364396a93429ac3ddff216d552523b0a6

**Comment by Tim-Cao** [comment]
ci:close

**Pull request closed without merging**